docs/testplan/PFC_Snappi_Additional_Testcases.md

```diff
@@ -0,0 +1,65 @@
+|           05 | CONGESTION               | **DETECT CONGESTION WITH EQUAL DISTRIBUTION OF LOSSLESS AND LOSSY TRAFFIC:**<br> Purpose of the testcase is to determine that amount of congestion faced by both lossless and lossy priority traffic. No losses for priority 3 and priority 4 traffic streams.                                                            | - Two 100Gbps Ingress and single 100Gbps egress <br>- Repeat test with 2x400Gbps ingress and 1x400Gbps egress <br>- IMIX packet-profile <br>- Equal traffic for all priorities                                                         | - 24% -  Priority 3 and 4 per ingress LC <br>- 36% Background flows [Priority 0, 1, and 2] per ingress LC <br>- Enabled PFCWD and credit-watchdog on ingress and egress.                                                                                               | - Single LC - Multiple ASIC <br>- Multiple LC                                    | 1. Packet drops for Lossy Traffic. <br>2. PFCs for lossless traffic. No packet drops for losslesss traffic. <br>3. Expected throughput of egress interface line-rate (Lossless + lossy traffic only) on egress interface of DUT. <br>4. Expected close to equal throughput for all priorities.                                                                                                                                                                                                 |
+|           06 | CONGESTION (MACSEC)      | **CONGESTION DETECTION WITH PAUSE FRAMES OVER MACSEC ENABLED LINK:** <br>Purpose of the test is to ensure that congestion is detected on DUT egress with MACSEC enabled links on ingress and egress and mix of lossless and lossy traffic. No drops for lossless priority traffic.                                        | - Single 100Gbps Ingress and Egress - PFC over MACSEC. <br> - Repeat with 400Gbps Ingress and Egress <br> - 1024B packet-size                                                                                                          | - 90% - Lossless Priority 3 and 4 per ingress LC <br>- 10% - Lossy Background Flow [Priority 0, 1, and 2] <br>- Disable PFCWD before the test <br>- Send PAUSE frames to egress interface of DUT.                                                                      | - Single LC - Multiple ASIC <br>- Single LC - Single ASIC <br>- Multiple LC      | 1. There should be NO packet drops on the DUT (IXIA Tx packets == IXIA Rx packets)<br>2. There should be PFCs on IXIA and DUT - Tx and Rx port. <br>3. No Lossless Priority traffic on IXIA Rx Interface. <br> 4. Only 10% lossy priority traffic allowed from the DUT. <br>5. No loss for Lossless traffic.                                                                                                                                                                                   |

```

Comment:
MACsec with PFC is not yet supported in SW/HW on Ixia. Looking for the feasibility of implementing this feature.